src/components/navbar/DropDown.tsx

```diff
@@ -1,33 +1,84 @@
-import { useAppDispatch } from "@/store/hooks";
+import { useAppDispatch, useAppSelector } from "@/store/hooks";
 import { clientSignOut } from "@/store/features/auth/authSlice";
 import { serverSignOut } from "@/app/(auth)/authService";
 
-export default function DropDown({ name }: { name: string }) {
+export default function DropDown({
+  name,
+  openState,
+}: {
+  name?: string;
+  openState?: boolean;
+}) { 
   const dispatch = useAppDispatch();
+  const [isOpen, setIsOpen] = useState(openState);
+  const currentVoyage = useAppSelector(
+    (state) => state.user.voyageTeamMembers[0]?.voyageTeam.name,
+  );
+  const menuRef = useRef<HTMLUListElement>(null);
+  const closed = "hidden";
+  const open =
+    "absolute z-[1] w-36 p-2 [&>*]:mt-2 mt-2 shadow bg-base-100 right-0 border border-neutral rounded-2xl";
 
+  const toggleMenu = () => {
+    setIsOpen(!isOpen);
+  };
+  //--------------------------------------------------------------------------------------
+  //'closeMenu' handleClickOutside' and 'useEffect' allow user to
+  //close menu on clicking anywhere but the menu.
+  const closeMenu = () => {
+    setIsOpen(false);
+  };
+  const handleClickOutside = (event: MouseEvent) => {
+    if (menuRef.current && !menuRef.current.contains(event.target as Node)) {
+      closeMenu();
+    }
+  };
+  useEffect(() => {
+    document.addEventListener("click", handleClickOutside);
+    return () => {
+      document.removeEventListener("click", handleClickOutside);
+    };
+  }, [isOpen]);
+  //-------------------------------------------------------------------------------------
   async function handleClick() {
     await serverSignOut();
     dispatch(clientSignOut());
   }
 
   return (
-    <div className="dropdown py-0 mx-2 dropdown-bottom">
+    <div className="relative py-0 mx-2">
       <label
         tabIndex={0}
-        className="btn m-0 p-0 bg-transparent border-none hover:border-none hover:bg-transparent text-base-300"
+        className="flex btn m-0 p-0 bg-transparent border-none hover:border-none hover:bg-transparent text-base-300"
+        onClick={toggleMenu}
       >
-        {name} <ChevronDownIcon className="w-4 text-base-300" />
+        {name}
+        <ChevronDownIcon className="w-4 text-base-300" />
       </label>
-      <ul
-        tabIndex={0}
-        className="dropdown-content z-[1] menu p-2 mt-2 shadow bg-base-100 right-0 border border-neutral rounded-2xl"
-      >
-        <DropDownLink title="Link 1" />
-        <DropDownLink title="404???" href="/hello404" />
-        <Button title="signout" type="button" onClick={handleClick}>
+      <ul tabIndex={0} className={isOpen ? open : closed} ref={menuRef}>
+        <li className="bg-secondary-content text-xs p-2 rounded-lg ">
+          My Status:
+          {/**Check if user is on voyage. If so, display voyage name/link*/}
+          {currentVoyage && <DropDownLink title={currentVoyage} />}
+          {!currentVoyage && (
```

Comment:
What is the advantage of using a negative conditional along with && operator? Could simplify this to just put the falsy value into a ternary in the above line (line 67).

Reply:
There isn't one! :blush: I've changed it.